Sources/DataSources/ASCollectionNode+Rx/ASCollectionSectionedDataSource.swift

```diff
@@ -36,7 +36,8 @@ open class ASCollectionSectionedDataSource<S: SectionModelType>: NSObject, ASCol
 
     fileprivate static func configureSupplementaryViewBlockNotSet(dataSource: ASCollectionSectionedDataSource<S>, node: ASCollectionNode, nodeForSupplementaryElementOfKind kind: String, indexPath: IndexPath) -> ASCellNodeBlock {
         // Users expect collectionNode(_nodeForSupplementaryElementOfKind:at:) will be executed in main thread according to api doc.
-        return { DispatchQueue.main.sync(execute: { dataSource.collectionNode(node, nodeForSupplementaryElementOfKind: kind, at: indexPath) }) }
+        let cellNode = dataSource.collectionNode(node, nodeForSupplementaryElementOfKind: kind, at: indexPath)
```

Comment:
Should we keep returning cell block inside `DispatchQueue.main` @foxware00 ?

Reply:
I wasn't sure why this was picked up in this PR, to be honest, but we merged this change a little while ago in #36. It's a valid change and I believe has been in 0.4.1 for a while now.

If you're on the main thread and call `main.sync` it'll deadlock

Reply:
@dangthaison91 looks like it's in `master` so I can't work out why it's coming up here

Reply:
@dangthaison91 update master and it's gone. Also bumped to `0.5.0`